### ISSUE TITLE
Graph related applications

### ISSUE THREAD

**Issue by joneswong** [ray-project/ray#9324]
### Building a graph related package based on Ray
Graph data are ubiquitous and thus have attracted more and more attention from both academia and industry.
Recently, KDD cup includes an AutoGraph track which encourages competitors to build a solution to automatically preprocess the graph data, do HPO, train graph neural networks.
We found that such an application is particularly suitable to be built upon Ray where
- the graph data would be maintained in the global storage
- many graph algorithm could be parallelized easily
- there are well-designed pytorch/tensorflow support for training graph neural networks
- there have been Ray Tune for doing HPO

So, is there any package in this line? If not, we are considering re-factoring our solution to the KDD cup as a graph-related package built upon Ray. Could you guys give us some suggestions? Thanks!

**Comment by rkooo567** [comment]
There was actually a guy who tried to do this from slack. I can connect you to him.

**Comment by simon-mo** [comment]
I'm not aware of any package in this line from the ray team. Definitely feel free to contribute! Is your solution it more about traditional graph algorithm or about graph neural net?

**Comment by joneswong** [comment]
> I'm not aware of any package in this line from the ray team. Definitely feel free to contribute! Is your solution it more about traditional graph algorithm or about graph neural net?

Both. We use some kinds of graph neural networks for node classification and, once there is no node feature, we first apply a random walk variant and then apply a decomposition algorithm to the proximity matrix for learning node features.
Thanks for your comment. We will have a try.

**Comment by rkooo567** [comment]
The feature request still relevant.

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
